apps/sharebymail/lib/ShareByMailProvider.php

```diff
@@ -405,6 +405,10 @@ protected function sendMailNotification($filename,
 		$emailTemplate->addHeader();
 		$emailTemplate->addHeading($this->l->t('%1$s shared »%2$s« with you', [$initiatorDisplayName, $filename]), false);
 		$text = $this->l->t('%1$s shared »%2$s« with you.', [$initiatorDisplayName, $filename]);
+		if ($expiration) {
+			$relativeDate = (string)$expiration->diff(new \DateTime('now'))->format("%a");
```

Comment:
```suggestion
			$relativeDate = $expiration->diff(new \DateTime())->format('%a');
```

`format` should always return a string.